### ISSUE TITLE
Off-by-one in progress tracker?

### ISSUE THREAD

**Issue by bhairavmehta95** [SciPhi-AI/sciphi#64]
**Describe the bug**
On some reruns for generating textbooks, I'm getting overlapping sections and off-by-one (i.e regen's the same last section). In addition, while I took it out for my code, restarting a run in between will always write the "This is an AI Generated..." in the textbook.

**To Reproduce**
Start a job, kill it halfway, watch where it restarts.

**Expected behavior**
I think the code is basically setting the `start_flag` too early (or the check + generation order should be switched).

Basically, right now, it regenerates the last section (i.e the sections described in `{textbook}_progress.txt`, but should actually generate the next one.

I will open this issue rn and investigate later today, keeping this issue updated and if needed, opening a PR to fix the issue. (I have some runs going, so don't want to disturb those)

**Comment by emrgnt-cmplxty** [comment]
hey @bhairavmehta95 - I removed the start_flag from the textbook generator since it was sloppy/untested.

Have you gotten a chance to run with the latest generator?
